[CS2113-W13-2] FitTrack

TheDinos opened the pull request:
FitTrack is a desktop application designed to assist students in training for the National Physical Fitness Award (NAPFA) test. It focuses on streamlining data entry through a Command Line Interface (CLI) for students who seek efficiency and focus.

Kobot7 commented on src/main/java/fittrack/calculator/Calculator.java:
```diff
@@ -0,0 +1,75 @@
+                }
+            }
+            return 0; // return 0 points as performance is below the minimum standard
+        } catch (InvalidAgeException e) {
```
Is the variable name "e" informative enough?

Kobot7 commented on src/main/java/fittrack/calculator/Calculator.java:
```diff
@@ -0,0 +1,75 @@
+    // Utility method to add age sub-table for a specific gender
+    protected static void addAgeSubTable(Map<LookUpKey, TreeMap<Integer, Integer>> pointsTable,
+                                         Gender gender, int age, int[][] points, boolean reverseOrder) {
+        assert age >= 12 && age <= 24 : "Age should be within 12 and 24 during table initialisation"; //
```
Accidental empty comment?

Kobot7 commented on src/main/java/fittrack/FitTrack.java:
```diff
@@ -0,0 +1,76 @@
+
+        // Until the exit command is entered, execute command then read user input
+        while (!input.equals(EXIT_COMMAND)) {
+            assert !input.trim().isEmpty() : "User input should not be null or empty";
```
Is this supposed to be an assertion or exception?

Kobot7 commented on src/main/java/fittrack/calculator/Calculator.java:
```diff
@@ -0,0 +1,75 @@
+        TreeMap<Integer, Integer> ageSubTable;
+        if (reverseOrder){
+            ageSubTable = new TreeMap<>(Comparator.reverseOrder());
+        } else{
```
Java reserved words should be followed by a white space

Kobot7 commented on src/main/java/fittrack/calculator/PullUpCalculator.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.TreeMap;
+
+public class PullUpCalculator extends Calculator {
+    private static final Map<LookUpKey, TreeMap<Integer, Integer>> pullUpTable = new HashMap<>();
+    private static final boolean reverseOrder = true;
```
Constant names must be all uppercase using underscore to separate words

Kobot7 commented on src/main/java/fittrack/exercisestation/PullUpStation.java:
```diff
@@ -0,0 +1,39 @@
+        return points;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
```
If implementation of getName() is the same for every child class, you may consider implementing it in the parent class instead

Kobot7 commented on src/main/java/fittrack/storage/Storage.java:
```diff
@@ -0,0 +1,108 @@
+
+public class Storage {
+
+    public static final String SAVE_FILE = "data/saveFile.txt"; // Path to the save file
+    public static final File SAVEFILE = new File(SAVE_FILE); // File object for the save file
```
Consider using more meaningful variable names, SAVE_FILE and SAVEFILE does not provide much context as to what differentiates them, leading to confusion when using it in the code

Kobot7 commented on src/main/java/fittrack/storage/Storage.java:
```diff
@@ -0,0 +1,108 @@
+        // Assert that the session list is not null before loading
+        assert sessionList != null : "Session list must not be null";
+
+        Scanner s = new Scanner(SAVEFILE); // Create a Scanner to read the save file
```
Use more informative variable names

Kobot7 commented on src/main/java/fittrack/parser/Parser.java:
```diff
@@ -0,0 +1,89 @@
+            assert description.contains(" ") : "Invalid user data format";
+            sentence = description.split(" ", 2);
+            user.setGender(sentence[0]);
+            user.setAge(sentence[1]);
+            break;
+        case ADD_SESSION_COMMAND:
+            assert !description.isEmpty() : "Session description must not be empty";
+            LocalDateTime currentTime = LocalDateTime.now();
+            sessionList.add(new TrainingSession(currentTime, description, user));
+            printAddedSession(sessionList);
+            break;
+        case EDIT_EXERCISE_COMMAND:
+            sentence = description.split(" ", 3);
+            assert sentence.length == 3 : "Edit exercise command requires exactly 3 arguments";
+            int sessionIndex = Integer.parseInt(sentence[0]) - 1;
+            int exerciseIndex = Integer.parseInt(sentence[1]);
+            String exerciseData = sentence[2];
+            assert sessionIndex >= 0 && sessionIndex < sessionList.size() : "Session index out of bounds";
+            sessionList.get(sessionIndex).editExercise(exerciseIndex, exerciseData);
+            break;
+        case LIST_SESSIONS_COMMAND:
+            printSessionList(sessionList); // Print the list of sessions
+            break;
+        case VIEW_SESSION_COMMAND:
+            int viewIndex = Integer.parseInt(description) - 1;
+            assert viewIndex >= 0 && viewIndex < sessionList.size() : "View session index out of bounds";
+            printSessionView(sessionList, viewIndex); // Print the session view
+            break;
+        case DELETE_SESSION_COMMAND:
+            int indexToDelete = Integer.parseInt(description) - 1;
+            assert indexToDelete >= 0 && indexToDelete < sessionList.size() : "Delete session index out of bounds";
+            TrainingSession sessionToDelete = sessionList.get(indexToDelete);
+            sessionList.remove(indexToDelete);
+            printDeletedSession(sessionList, sessionToDelete);
+            break;
+        default:
+            printUnrecognizedInputMessage(); // Response to unrecognized inputs
+            break;
+        }
+    }
```
Try to avoid long methods. Perhaps consider extracting out the commands into separate methods

Kobot7 commented on src/main/java/fittrack/parser/Parser.java:
```diff
@@ -0,0 +1,280 @@
+                    }
+                }
+                Goal newGoal = new Goal(goalDescription, goalDeadline);
+                goalList.add(newGoal);
+                printAddedGoal(goalList);
+            } else {
+                System.out.println("Please specify a goal to add.");
+            }
+            break;
+
+        case "delete-goal":
+            try {
+                int index = Integer.parseInt(description) - 1;
+                if (index >= 0 && index < goalList.size()) {
+                    goalList.remove(index);
+                    System.out.println("Goal at index " + (index + 1) + " has been removed.");
+                } else {
+                    System.out.println("Invalid goal index.");
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Please specify a valid index to delete.");
+            }
+            break;
+
+        case "list-goal":
+            if (goalList.isEmpty()) {
+                System.out.println("No goals to display.");
+            } else {
+                System.out.println("Goals:");
+                for (int i = 0; i < goalList.size(); i++) {
+                    System.out.println((i + 1) + ". " + goalList.get(i));
+                }
+            }
+            break;
+
+        default:
+            printUnrecognizedInputMessage(); // Response to unrecognized inputs
+            break;
+        }
+    }
```
Method is too long. Do consider extracting out code for each command to a separate method

Kobot7 commented on src/main/java/fittrack/parser/Parser.java:
```diff
@@ -0,0 +1,280 @@
+     *
+     * @param inputDeadline A string input by the user. Intended format is DD/MM/YYYY or DD/MM/YYYY HH:mm:ss.
+     * @return A {@code LocalDateTime} object indicating reminder deadline
+     * @throws IllegalArgumentException
```
Missing description for exception

Kobot7 commented on src/main/java/fittrack/parser/Parser.java:
```diff
@@ -0,0 +1,280 @@
+            // Try to parse with time (dd/MM/yyyy HH:mm:ss)
+            try {
+                return LocalDateTime.parse(inputDeadline, dateTimeFormatter);
+            } catch (DateTimeParseException e) {
```
Use more descriptive variable names, it is unclear what "e" refers to

Kobot7 commented on src/main/java/fittrack/reminder/Reminder.java:
```diff
@@ -0,0 +1,45 @@
+
+    public void printReminderDescription() {
+        System.out.print(this.reminderDescription + " | " +
+                this.reminderDeadline.format(DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm")) + System.lineSeparator());
```
Line length should be no longer than 120 chars

Kobot7 commented on src/main/java/fittrack/fitnessgoal/AddFitnessGoal.java:
```diff
@@ -0,0 +1,18 @@
+
+    public void addGoal(User user) {
+        user.addGoal(goal); // Add the goal object to the user's goal list
+        // Use the correct method to get the description of the goal
```
Comments should explain the WHAT and WHY aspects of the code, rather than the HOW aspect. In this case, a comment may not be necessary

Kobot7 commented on src/main/java/fittrack/fitnessgoal/Goal.java:
```diff
@@ -0,0 +1,28 @@
+    private final String description;
+    private final LocalDateTime deadline;
+
+    // Constructor
```
Would be good to follow Javadoc format to document methods instead

Kobot7 commented on src/main/java/fittrack/fitnessgoal/Goal.java:
```diff
@@ -0,0 +1,28 @@
+        return deadline;
+    }
+
+    // Optionally, you might want to override the toString() method for easy printing
```
Was this comment intended?

Kobot7 commented on src/main/java/fittrack/parser/Parser.java:
```diff
@@ -0,0 +1,280 @@
+            printUpcomingReminders(reminderList);
+            break;
+
+        case "add-goal":  // use "add-goal" consistently in input and command handling
+            if (!description.isEmpty()) {
+                String[] goalParts = description.split(" ", 2);
+                String goalDescription = goalParts[0];
+                LocalDateTime goalDeadline = null;
+
+                if (goalParts.length > 1) {
+                    String goalDeadlineInput = goalParts[1];
+                    try {
+                        goalDeadline = parseGoalDeadline(goalDeadlineInput);
+                    } catch (IllegalArgumentException e) {
+                        System.out.println("Invalid date format: " + e.getMessage());
+                        return;
+                    }
+                }
+                Goal newGoal = new Goal(goalDescription, goalDeadline);
+                goalList.add(newGoal);
+                printAddedGoal(goalList);
+            } else {
+                System.out.println("Please specify a goal to add.");
+            }
+            break;
```
Avoid deep nesting

Kobot7 commented on src/main/java/fittrack/parser/Parser.java:
```diff
@@ -0,0 +1,280 @@
+            String inputDeadline = sentence[1];
+            description = sentence[0];
+
+            assert !description.isEmpty() : "Reminder description must not be empty";
```
Is this intended to be an assertion or exception?

shyaamald commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,77 @@
 ## Design & implementation
 
+# High Level Functionalities
+![HighLevel.png](HighLevel.png)
```
Are the arrows standard?
![image](https://github.com/user-attachments/assets/8fc1e66f-5634-4f89-815a-b9257ac5626f)

shyaamald replied:
Could the roles be closer to the arrow?

rharwo replied:
Somewhat complex class diagram. Maybe split into components?

Progresst-8 commented on docs/TrainingSessionEditState.png:
What is the relation between ExerciseStation class and the 6 station classes below? Is there some kind of inheritance?

The class diagram has non-standard notation such as: 
visibility ![image](https://github.com/user-attachments/assets/c201fe02-9810-40af-a897-378c3cb29f3b)
class label ![image](https://github.com/user-attachments/assets/c200122d-51c7-4e78-bad5-4d31785ac486)
arrowhead ![Screenshot 2024-10-30 121253](https://github.com/user-attachments/assets/62e608ad-d310-48e3-97f0-5486d19698db)

shyaamald commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,77 @@
+
+The following sequence diagram illustrates the function calls involved in this process:
+
+![editExerciseSequenceDiagram.png](editExerciseSequenceDiagram.png)
```
Stickman is not standard here
![image](https://github.com/user-attachments/assets/a6cde043-7364-40ef-99a2-77c9e573c2e8)

shyaamald replied:
Could activation bars be added?
![image](https://github.com/user-attachments/assets/28c1486a-31d8-47d3-a3c4-6494d6edc35c)

rharwo replied:
I believe doesn't follow correct format for class name. I think it should be ":ClassName"
![image](https://github.com/user-attachments/assets/06bf4be2-d672-4327-9a46-5f94fdcd6ee0)

rharwo replied:
Should Parser be shown as an object? i.e. should it be in a box?
![image](https://github.com/user-attachments/assets/8a3fba1d-1d94-4276-809a-d1deff81e1ad)

rharwo replied:
Should activation bars be either added here or removed in the next sequence diagram to follow a standard format?

Progresst-8 commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,77 @@
+
+![editExerciseSequenceDiagram.png](editExerciseSequenceDiagram.png)
+
+Additionally, the state diagram below shows the end state of the `editExercise` function after execution of the command,
```
The state diagram is a specific kind of UML behavioral diagram. 

> Additionally, the state diagram below ...

I don't think that's what you're referring to here, as this seems like structural?
Additionally, is this a class diagram or an object diagram? It contains the non-underlined name (like a class diagram) and values of variables (like an object diagram)

Progresst-8 commented on docs/editExerciseSequenceDiagram.png:
Why is the Parser class represented by a stickfigure? Doesn't the stickfigure represent a user, not a class?

The instance name is missing the colon before it, as per Sequence diagram conventions.

Why are the activation bars missing?

Does Parser have a custom-defined print method, since print() is pointing to Parser? If so, why is a Parser printing (writing to output), when it should be parsing (reading from input)?

shyaamald commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,77 @@
+   to points based on their age and gender. The points are returned to the exercise station, where they are stored.
+
 
 ## Product scope
```
![image](https://github.com/user-attachments/assets/71c37a4b-e74a-4d6b-9620-0d56911b6f35)
Could add own user stories

Progresst-8 commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,77 @@
+
+3. **Calculator Logic**: The calculator class uses a lookup table, which maps the user's performance 
+   to points based on their age and gender. The points are returned to the exercise station, where they are stored.
+
```
Are the user stories below still unchanged from the template?
Where are the user stories for your project?

Progresst-8 commented on docs/HighLevel.png:
Is this supposed to be a high-level overview of your entire program? Inclusion of low-level implementation details seems counter to what an overview is supposed to do.
If so, isn't a class diagram not appropriate for this purpose?
Perhaps this should be an architecture diagram instead of a class diagram?

Additionally, for a class diagram, labels require the arrowhead, which is not present here.
![image](https://github.com/user-attachments/assets/9f3f5b2b-d98e-432f-80b0-4d950cbc14d3)
![image](https://github.com/user-attachments/assets/d828ca86-553b-40b2-9b33-93cfcb532f72)

shyaamald commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,77 @@
+Additionally, the state diagram below shows the end state of the `editExercise` function after execution of the command,
+`editExercise(Exercise.PULL_UP, 1)`:
+
+![TrainingSessionEditState.png](TrainingSessionEditState.png)
```
![image](https://github.com/user-attachments/assets/20635339-e35e-426d-bb5e-8c216767cc03)
This is not a state diagram

Progresst-8 reviewed (commented):
DG looks mostly okay, but some amendments are required.
Some parts of the DG are still left unchanged from the template. Those should be updated to reflect your project.

rharwo commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,77 @@
+initializes an EnumMap, which populates the various `ExerciseStation` instances with their initial values. 
+Below is a representation of how the `ExerciseStations` are initialized:
+
+![TrainingSessionInitialState.png](TrainingSessionInitialState.png)  
```
Unclear what the visibility is in this class diagram
![image](https://github.com/user-attachments/assets/073c9b53-b9c0-4456-82d4-ba69a972b2bd)

rharwo commented on docs/DeveloperGuide.md:
```diff
@@ -6,8 +6,77 @@
+method. The main responsibility of this method is to invoke the `calculatePoints()` function from the 
+respective **calculator** class (e.g., `PullUpCalculator`, `SitUpCalculator`), which holds the points calculation logic.
+
+![getPointsSequenceDiagram.png](getPointsSequenceDiagram.png)
```
Should sequence diagram return arrows be labeled "return {variable}" or just "variable"?
![image](https://github.com/user-attachments/assets/b25dd462-507d-42a0-a8fc-e9f1e1518500)

rharwo replied:
Could the "PullUpStation" and "PullUpCalculator" be named to be generic names to showcase what is written in the description?

DucPhong135 commented on docs/DeveloperGuide.md:
This diagram is abit hard to look at
![image](https://github.com/user-attachments/assets/4b5939f3-18ff-4f6b-8797-b5a109131493)

DucPhong135 commented on docs/DeveloperGuide.md:
The relationship label can be placed uniformly for better layout
![Uploading image.png…]()

DucPhong135 commented on docs/DeveloperGuide.md:
the use of the uses label on all associations might be redundant, as it doesn’t add much information and could be removed for a cleaner look
![Uploading image.png…]()

DucPhong135 commented on docs/DeveloperGuide.md:
If Calculator is intended to be a shared utility, can the relationship between Calculator and ExerciseStation changed to a dependency that is accessed via FitTrack or TrainingSession instead?
![image](https://github.com/user-attachments/assets/6ffacb15-66df-475f-bcf4-6d885128229a)

DucPhong135 commented on docs/DeveloperGuide.md:
Overall, really well structured DG, diagrams really clarify the functionalities of the app

Zackermax committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-2/tp into 205-docs-update-dg-for-v20

# Conflicts:
#	docs/DeveloperGuide.md

Yvorm committed:
Merge branch 'master' of https://github.com/Yvorm/tp

# Conflicts:
#	FitTrackLogger.log

Zackermax committed:
Merge pull request #228 from Zackermax/205-docs-update-dg-for-v20

205 docs update dg for v20

Yvorm committed:
Merge branch 'master' of https://github.com/Yvorm/tp

CheeKiangg committed:
fix: correct return arrows in addTrainingSession puml diagram

CheeKiangg committed:
feat: add GraphPerformanceTime class section in developer guide

Zackermax committed:
Docs: Update DG for V2.0

Zackermax committed:
Merge pull request #230 from CheeKiangg/Branch-UpdateDG

Branch: Update DG

TheDinos committed:
Update DG for graph features.

TheDinos committed:
Merge branch 'master' into 205-docs-update-dg-for-high-level-functionalities

CheeKiangg committed:
Merge pull request #231 from Zackermax/205-docs-update-dg-for-v20-2

Docs: Update DG for V2.0

TheDinos committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-2/tp into 205-docs-update-dg-for-high-level-functionalities

CheeKiangg committed:
Merge pull request #232 from TheDinos/205-docs-update-dg-for-high-level-functionalities

205 docs update dg for high level functionalities

TheDinos committed:
Update minor error in DG.

TheDinos committed:
Implement graphing functionality for user input.

TheDinos committed:
Fix checkstyle.

TheDinos committed:
Update checkstyle.

TheDinos committed:
Merge pull request #234 from TheDinos/233-implement-user-input-for-graphing-functionalities

233 implement user input for graphing functionalities

Yvorm committed:
Merge remote-tracking branch 'origin/update-UserGuide-V2' into dg-ug-contribution-update

CheeKiangg committed:
feat: update DG
- include functionality of Calculator class
- include PUML diagram for graphPerformanceTime

CheeKiangg committed:
feat: add PUML diagram for time based station graph

CheeKiangg committed:
refactor: rename graphExerciseTime to generateMainGraphPerformance for consistency

CheeKiangg committed:
fix: correct checkstyle.

TheDinos committed:
Merge pull request #235 from CheeKiangg/Branch-UpdateDGv2

Branch updateDGv2

TheDinos committed:
Update point graphs functionality for UG.

TheDinos committed:
Fix minor ug mistakes.

CheeKiangg committed:
Merge pull request #236 from TheDinos/207-update-ug-with-final-v20-functionality

Update point graphs functionality for UG.

CheeKiangg committed:
feat: add Display Performance Graph
- For non-time based station, bar graph of raw performance metric against session index is displayed
- For time based station, scatter graph of normalised performance metric against session index is displayed

CheeKiangg committed:
refactor: ensure consistency in the itle of FitTrackCLI's Features

TheDinos committed:
Merge pull request #237 from CheeKiangg/Branch-UpdateUG

Branch update ug

TheDinos committed:
Update grade file to compile jar file.

TheDinos committed:
Fix graph points parser bug.

Yvorm committed:
Removed Unneeded Classes

CheeKiangg committed:
update individual contribution

Zackermax committed:
Docs: Update DG for V2.0

Zackermax committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-2/tp into 205-docs-update-dg-for-v20-3

CheeKiangg committed:
Merge pull request #241 from Zackermax/205-docs-update-dg-for-v20-3

205 docs update dg for v20 3

Zackermax committed:
Merge pull request #240 from CheeKiangg/Branch-IndividualContribution

update individual contribution

CheeKiangg committed:
Feat: add comments for functions in GraphPerformance and GraphPerformanceTime class

TheDinos committed:
Update build.gradle

Zackermax committed:
Merge pull request #239 from TheDinos/238-fix-bug-in-parser-for-graph-points

238 fix bug in parser for graph points

CheeKiangg committed:
fix: correct checkstyle

CheeKiangg committed:
Merge pull request #243 from CheeKiangg/Branch-AddCommentsForGraphPerformanceTime

Feat: add comments for functions in GraphPerformance and GraphPerformanceTime

CheeKiangg committed:
Update UG and DG with link to NAPFA score table

CheeKiangg committed:
Merge pull request #245 from CheeKiangg/Branch-InsertLinkToNAPFATable

Update UG and DG with link to NAPFA score table

Yvorm committed:
Remove Unneeded Classes

Yvorm committed:
Add initial Save-File initialising & Parsing support

Yvorm committed:
Update parsing logic

Yvorm committed:
Remove unneeded methods

Yvorm committed:
Rename misc commands

Yvorm committed:
Reimplement FoodEntry as Saveable subclass

Yvorm committed:
Implement SaveFile Functionality for Food & Water Entries

Yvorm committed:
Add Initial Changes to DG

Yvorm committed:
Merge branch 'master' of https://github.com/Yvorm/tp

Yvorm committed:
Merge branch 'update-UserGuide-ReminderFunctionality' into food-water-storage-integration

Zackermax committed:
UG Help Function Typo Fix

Zackermax committed:
Merge pull request #246 from AY2425S1-CS2113-W13-2/UG-Help-Function-Typo-Fix

UG Help Function Typo Fix

Yvorm committed:
Update DG and UG with Storage / Reminder / Water / Food function descriptions

Yvorm committed:
Add missing classes

Yvorm committed:
Fix CI failures

Zackermax committed:
Merge pull request #247 from Yvorm/food-water-storage-integration

Food water storage integration + DG + UG

Yvorm committed:
Add DateTime parsing error messages

Yvorm committed:
Add JUnit testing for DateTimeParsing

Yvorm committed:
Update DeleteReminder Index validation

Yvorm committed:
Abstract Parser Helper Functions

Zackermax committed:
Parser now accepts lowercase exercise acronyms

Zackermax committed:
Merge pull request #309 from Zackermax/294-pe-dtester-d-error-message-does-not-match-the-error-when-entering-sit-up-value

Parser now accepts lowercase exercise acronyms

Zackermax committed:
Explicitly list exercise acronyms in userguide and help function

Zackermax committed:
Updated sequence diagrams

Yvorm committed:
Fix UserUpdate Logic

Yvorm committed:
Update Reminder-related functions + UG to use "//" as date/time marker

Yvorm committed:
Update missed UserUpdate function

Yvorm committed:
Add reminder class getter functions

Yvorm committed:
Add JUnit tests for Reminder-Relation input parsing

Yvorm committed:
Add general JUnit test framework for Parsing testing

Yvorm committed:
Correct reminder print function names

Yvorm committed:
Update Parsing JUnit framework

Yvorm committed:
Merge branch 'master' into fix-reminder-bugs

# Conflicts:
#	src/main/java/fittrack/FitTrack.java
#	src/main/java/fittrack/messages/Messages.java

Zackermax committed:
Merge pull request #310 from Zackermax/263-pe-dtester-c-exercise-acronyms-not-explicitly-listed-out

Explicitly list exercise acronyms in userguide and help function

Yvorm committed:
Fix CI failures

Zackermax committed:
Created modify DateTime command, functionality and documentation

Zackermax committed:
Updated PPP

Zackermax committed:
Updated User Stories

Zackermax committed:
Update PPP name

ayushi0803 committed:
exception catching for date time format for reminders

ayushi0803 committed:
checkstyle

ayushi0803 committed:
documentation suggestions

ayushi0803 committed:
goal line separator added

ayushi0803 committed:
Standardization of Visual Output Bordering for Error Input

ayushi0803 committed:
fitnes goal text - editing the goal to pass the test with deadline

ayushi0803 committed:
fitness goal test

ayushi0803 committed:
fixing the add-goal ex[pected outputs

ayushi0803 committed:
fitness goal test

ayushi0803 committed:
Merge branch 'goal-line-separator' of github.com:ayushi0803/tP into goal-line-separator

* 'goal-line-separator' of github.com:ayushi0803/tP:
  fitness goal test
  fixing the add-goal ex[pected outputs

ayushi0803 committed:
checkstyle

TheDinos committed:
Fix formatting in TrainingSession.

TheDinos committed:
Update checking of valid inputs for shuttle run and walk and run.

TheDinos committed:
Add JUnit tests for valid user inputs of walk and run and shuttle run.

TheDinos committed:
Merge pull request #312 from Zackermax/307-pe-dtester-c-add-session-function-does-not-allow-for-datetime-input-or-editing

307 pe dtester c add session function does not allow for datetime input or editing

CheeKiangg committed:
feat: Update FAQ in User Guide

CheeKiangg committed:
Merge pull request #316 from TheDinos/253-pe-dtester-e-error-handling-on-walk-and-run-timing-input

Update invalid input handling for Walk and Run and Shuttle Run

CheeKiangg committed:
Merge branch 'master' into Branch-UserGuideFAQ

TheDinos committed:
Update UG with summarised table of formats for all exercises.

TheDinos committed:
Merge pull request #317 from CheeKiangg/Branch-UserGuideFAQ

feat: Update FAQ in User Guide

CheeKiangg committed:
Merge pull request #318 from TheDinos/305-pe-dtester-c-edit-function-for-shuttle-run-rejects-integer-input

Update UG with summarised table of formats for all exercises.

TheDinos committed:
Add missing commands gpoints and gperformance to help command in both UG and messages class.

Zackermax committed:
Merge pull request #319 from TheDinos/304-pe-dtester-d-help-command-does-not-do-what-it-says-in-user-guide

Add missing commands gpoints and gperformance to help command.

Zackermax committed:
Merge pull request #315 from ayushi0803/goal-line-separator

Goal line separator

CheeKiangg committed:
add Modify command in summary

TheDinos committed:
Update UG to reflect changes in modify command.

ayushi0803 committed:
user guide documentation update on goals (delete and list)

Zackermax committed:
Merge pull request #322 from CheeKiangg/Branch-addmodifycommand

add Modify command in summary

ayushi0803 committed:
remove example from list-goal

ayushi0803 committed:
goal commenting

ayushi0803 committed:
health profile comments

TheDinos committed:
Implement reordering of trainingsession objects when datetime is changed.

Update session datetime format.

TheDinos committed:
Implement JUnit tests to check for reordering of trainingsessions.

TheDinos committed:
Update JUnit function name for checkstyle.

ayushi0803 committed:
parser comments

ayushi0803 committed:
fit track and user comments

ayushi0803 committed:
calculator comments

ayushi0803 committed:
pull up calc comments

ayushi0803 committed:
calculator comments

ayushi0803 committed:
checkstyle errors

Zackermax committed:
Merge pull request #325 from TheDinos/307-pe-dtester-c-add-session-function-does-not-allow-for-datetime-input-or-editing

Update modify command logic.

Zackermax committed:
Merge pull request #326 from ayushi0803/comments-contributions-

Comments contributions

ayushi0803 committed:
Ayushi.md File updation

ayushi0803 committed:
ayushi.md file contribution updation

ayushi0803 committed:
rename file

Yvorm committed:
Merge branch 'master' into fix-reminder-bugs

# Conflicts:
#	src/main/java/fittrack/messages/Messages.java
#	src/main/java/fittrack/parser/Parser.java
#	src/test/java/fittrack/parser/ParserTest.java

Yvorm committed:
Fix JUnit test broken by merge

Yvorm committed:
Correct CI issues in JUnit Tests

Yvorm committed:
Correct CI issues in JUnit Tests

Yvorm committed:
Merge branch 'fix-reminder-bugs' of https://github.com/Yvorm/tp into fix-reminder-bugs

Yvorm committed:
Final CI pass

Zackermax committed:
Merge pull request #327 from ayushi0803/Contributions-

Contributions - Ayushi

Zackermax committed:
Merge pull request #311 from Yvorm/fix-reminder-bugs

Fix reminder bugs

Zackermax committed:
Update contribution page and JUnit tests

Yvorm committed:
Add rejection of vertical bar input and JUnit testing

Yvorm committed:
Add user initialisation from SaveFile

Yvorm committed:
Add data_delimiter and data_delimiter_regex to standardise input parsing

Yvorm committed:
Update UpdateSaveFile methods in program to match new params

Zackermax committed:
CI fix

Zackermax committed:
Merge pull request #329 from Yvorm/fix-SaveFile-bugs

Fix SaveFile Crashes & Implement User initialisation from SaveFile

Zackermax committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-2/tp into 210-finish-contribution-page

Zackermax committed:
CI fix

Zackermax committed:
Merge pull request #328 from Zackermax/210-finish-contribution-page

Update contribution page and JUnit tests

Zackermax committed:
Fixed user initialisation

TheDinos committed:
Merge pull request #330 from Zackermax/210-finish-contribution-page-1

Fixed user initialisation

Zackermax committed:
Parser accepts uppercase inputs

ayushi0803 committed:
health profile tests

TheDinos committed:
Merge pull request #332 from Zackermax/331-accept-uppercase-commands-in-parser

Parser accepts uppercase inputs

ayushi0803 committed:
food entry test

ayushi0803 committed:
removing unused imports

ayushi0803 committed:
checkstyle

ayushi0803 committed:
checkstyle fixes

ayushi0803 committed:
contribution update

Zackermax committed:
Merge pull request #333 from ayushi0803/J-unit

health profile tests

Zackermax committed:
New JUnit Tests for Ui

Zackermax committed:
CI fix

Zackermax committed:
Merge pull request #337 from Zackermax/335-junit-tests-80-line-coverage

New JUnit Tests for Ui

ayushi0803 committed:
add fitness goal test and food intake test full coverages

TheDinos committed:
Add JavaDocs for TrainingSession.

TheDinos committed:
Add JUnit tests for TrainingSession class.

ayushi0803 committed:
checkstyle

ayushi0803 committed:
import removed

ayushi0803 committed:
indentation

TheDinos committed:
Fix checkstyle.

CheeKiangg committed:
Adhere to coding standard
- Remove obvious and incorrect comments description.

TheDinos committed:
Merge pull request #342 from CheeKiangg/Branch-EnsureCodeQuality

Adhere to coding standard

CheeKiangg committed:
Merge pull request #340 from TheDinos/339-update-javadoc-and-junit-tests-for-trainingsession-class

Update javadoc and junit tests for trainingsession class

TheDinos committed:
Update UG with more information about set user usage.

Update UG from minor flaws.

TheDinos committed:
Merge pull request #344 from TheDinos/343-update-ug

Update UG with more information about set user usage.

Yvorm committed:
Fix crashes related to input containing "|" character

Yvorm committed:
Merge branch 'master' into update-JUnit-tests

Yvorm committed:
Add reminder class JUnit tests

CheeKiangg committed:
feat: add test class for GraphPerformanceTime

TheDinos committed:
Update DG.

CheeKiangg committed:
fix ci

TheDinos committed:
Update DG.

TheDinos committed:
Merge branch 'master' into 346-update-dg

TheDinos committed:
Merge pull request #345 from CheeKiangg/Branch-GraphPerformanceTimeTestClass

feat: add test class for GraphPerformanceTime

CheeKiangg committed:
Merge pull request #347 from TheDinos/346-update-dg

Update DG.

CheeKiangg committed:
Feat: update PPP

CheeKiangg committed:
Merge pull request #349 from CheeKiangg/PPP

Feat: update PPP

TheDinos committed:
Update thedinos ppp.

TheDinos committed:
Update minor errors in thedinos ppp.

CheeKiangg committed:
Merge pull request #350 from TheDinos/348-update-contribution-page

Update contribution page for thedinos.

CheeKiangg committed:
display sequence diagram below workflow

TheDinos committed:
Merge pull request #352 from CheeKiangg/FormatDGv2.1

display sequence diagram below workflow

Yvorm committed:
Fix bug with ParserTest logic

Yvorm committed:
Add Storage JUnit Tests

Yvorm committed:
Edit Reminder JUnit logic

Zackermax committed:
Merge pull request #338 from ayushi0803/testing

add fitness goal test and food intake test full coverages

Yvorm committed:
Extend Storage JUnit Coverage

Yvorm committed:
Merge branch 'master' into update-JUnit-tests

Zackermax committed:
Updated PPP

Zackermax committed:
Merge pull request #353 from Yvorm/update-JUnit-tests

Extend JUnit test coverage of Reminder and Storage Classes

Zackermax committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-2/tp into 348-update-contribution-page

CheeKiangg committed:
Merge pull request #354 from Zackermax/348-update-contribution-page

Updated PPP

TheDinos committed:
Initial Junit test files

TheDinos committed:
Merge branch 'master' into 335-junit-tests-graph

Zackermax committed:
General bugfixes

CheeKiangg committed:
Merge pull request #355 from Zackermax/335-junit-tests-80-line-coverage-2

General bugfixes

Yvorm committed:
Add Reminder-Related-Parser JUnit tests

Yvorm committed:
Fix logic error in TrainingSession toSaveString method

TheDinos committed:
Add JUnit tests for GraphPoints.

TheDinos committed:
Add JUnit tests for GraphPerformance.

Yvorm committed:
Fix modifySession method savefile + printing functionality

Yvorm committed:
Merge branch 'master' into add-ReminderParser-JUnit-tests

TheDinos committed:
Add Javadocs to Graph classes.

CheeKiangg committed:
update PPP

TheDinos committed:
Fix checkstyle.

CheeKiangg committed:
Merge pull request #359 from CheeKiangg/PPP-finalised

update PPP

TheDinos committed:
Fix checkstyle.

TheDinos committed:
Merge pull request #356 from Yvorm/add-ReminderParser-JUnit-tests

Add reminder parser JUnit tests & Fix logic error in TrainingSession savedata

TheDinos committed:
Merge pull request #357 from TheDinos/335-junit-tests-graph

Add JUnit tests and Javadoc for Graph classes.

TheDinos committed:
Fix formatting for table.

TheDinos committed:
Update PPP.

CheeKiangg committed:
remove duplicate

TheDinos committed:
Merge pull request #360 from TheDinos/358-final-update-to-ppp

Fix formatting for table.

Yvorm committed:
Updated PPP

CheeKiangg committed:
Merge branch 'PPP-finalisedversion' into FinalPPP

CheeKiangg committed:
remove duplicate

TheDinos committed:
Update thedinos ppp.

CheeKiangg committed:
Merge pull request #362 from Yvorm/update-PPP-AvjayBhar

Updated PPP

CheeKiangg committed:
Merge pull request #361 from CheeKiangg/FinalPPP

Final ppp

TheDinos committed:
Merge pull request #363 from TheDinos/358-final-update-to-ppp2

Update thedinos ppp.

TheDinos committed:
Merge branch 'master' into FinalPPPv2.1

CheeKiangg committed:
Merge pull request #364 from CheeKiangg/FinalPPPv2.1

remove duplicate

Yvorm committed:
Update UG

Yvorm committed:
Merge branch 'master' into update-UG

CheeKiangg committed:
shorten description

CheeKiangg committed:
Merge pull request #365 from CheeKiangg/PPP_v

shorten description

TheDinos committed:
Add PPP

TheDinos committed:
Update README to reflect UG.

CheeKiangg committed:
Merge pull request #366 from TheDinos/358-final-update-to-readme

Update README to reflect UG.

CheeKiangg committed:
Add files via upload

Yvorm committed:
Add DG diagrams / UML

Yvorm committed:
Update Reminder-related DG entries

Yvorm committed:
Update Storage Class diagram

Yvorm committed:
Add misc Updates to DG

Yvorm committed:
Update Documentation

Zackermax committed:
Refactor PPP

Zackermax committed:
Refactor PPP

Zackermax committed:
Merge pull request #368 from Yvorm/update-UG-DG

Update ug dg

Zackermax committed:
Merge branch 'master' of https://github.com/AY2425S1-CS2113-W13-2/tp into 367-refactor-ppp-name

CheeKiangg committed:
Merge pull request #369 from Zackermax/367-refactor-ppp-name

Zackermax committed:
Refactor AboutUs

Zackermax committed:
Merge pull request #371 from Zackermax/370-refactor-aboutus

Refactor AboutUs
